misk-policy-testing/src/main/kotlin/misk/policy/opa/OpaTestPlugin.kt

```diff
@@ -0,0 +1,81 @@
+package misk.policy.opa
+
+import com.github.dockerjava.api.model.Bind
+import com.github.dockerjava.api.model.Binds
+import com.github.dockerjava.api.model.Frame
+import com.github.dockerjava.api.model.HostConfig
+import com.github.dockerjava.api.model.Volume
+import com.github.dockerjava.core.DockerClientBuilder
+import com.github.dockerjava.core.async.ResultCallbackTemplate
+import com.github.dockerjava.core.command.WaitContainerResultCallback
+import com.github.dockerjava.netty.NettyDockerCmdExecFactory
+import okio.Buffer
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.provider.Property
+import org.gradle.kotlin.dsl.create
+
+interface OpaTestConfig {
+  val policyDir: Property<String>
+}
+
+class OpaTestPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    val config: OpaTestConfig = project.extensions.create("opa")
+
+    val opaTestWithDocker = project.tasks.create("opaTestWithDocker") {
+      group = "OPA"
+      description = "Evaluate OPA tests via docker OPA instance"
+
+      doLast {
+        val dockerClient = DockerClientBuilder.getInstance()
+          .withDockerCmdExecFactory(NettyDockerCmdExecFactory())
+          .build()
+        val policyDir = project.projectDir.absolutePath + "/" + config.policyDir.get()
+        val containerId = dockerClient.createContainerCmd("openpolicyagent/opa")
+          .withCmd(listOf("test", "/repo", "-b", "--explain", "fails"))
+          .withHostConfig(
+            HostConfig.newHostConfig()
+              .withAutoRemove(true)
+              .withBinds(Binds(Bind(policyDir, Volume("/repo"))))
+          )
+          .withName("opa_tester")
+          .withTty(true)
+          .exec().id
+
+        dockerClient.startContainerCmd(containerId)
```

Comment:
Should also wait here until the docker has downloaded the docker image. Possible also consider pinning the image version